src/SlmQueueBeanstalkd/Worker/BeanstalkdWorker.php

```diff
@@ -28,10 +29,16 @@ public function processJob(JobInterface $job, QueueInterface $queue)
         }
 
         try {
-            $job->execute();
+            $result = $job->execute();
             $queue->delete($job);
+
+            if (null !== $result) {
+                return $result;
+            }
+            return WorkerEvent::JOB_STATUS_SUCCESS;
         } catch (Exception $exception) {
             $queue->bury($job, array('priority' => Pheanstalk::DEFAULT_PRIORITY));
```

Comment:
I created a local change in my project, similar to your code.

Analyzing the workflow, I receive some exceptions showing that beanstalkd don't know the job. Thinking about that, there's a first bury at line 40 and a second bury in another point of the code, because we return a `WorkerEvent::JOB_STATUS_FAILURE`. Am I right? :confused: